tests/goto_wcc_invalid_test.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+set -o pipefail
+
+source ./test_sub.sh
+
+function test_backward_goto() {
+  begin_test_suite "Backward Goto"
+  
+  compile_error 'simple backward goto' '
+int main() {
+start_label:
```

Comment:
In fact, it does not determine that you are jumping backwards,
so if you insert `if` statement in front of the label, then compile successfully.
This is silly example but causes wrong result.

Reply:
Yes, currently we do no verify that the label was traversed when we traverse goto. I'll fix that.

Reply:
Fixed.